plugin/pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -159,3 +162,26 @@ func (meta *predicateMetadata) AddPod(addedPod *v1.Pod, nodeInfo *schedulercache
 	}
 	return nil
 }
+
+// ShallowCopy copies a metadata struct into a new struct and creates a copy of
+// its maps and slices, but it does not copy the contents of pointer values.
+func (meta *predicateMetadata) ShallowCopy() algorithm.PredicateMetadata {
```

Comment:
probably needs tests to catch drift if the struct changes

Reply:
+1